engine-api/src/main/java/pl/edu/icm/unity/engine/api/GroupsManagement.java

```diff
@@ -32,6 +32,13 @@
 	 */
 	void addGroup(Group toAdd) throws EngineException;
 
+	/**
+	 * Adds a new group in a recursively way
+	 * @param toAdd group to add
+	 * @param recursive flag to add group recursively
```

Comment:
After some thinking I guess "recursive" is not a right term here. This is rather "WithParents".

Also note that the API should not be developed like this: boolean method args are considered generally an anti pattern. Instead you can have two methods, named properly, without a boolean param at all. There are sometimes practical exceptions, but that can be considered a rule of thumb.

Reply:
done